cli/src/commands/publish.js

```diff
@@ -0,0 +1,216 @@
+            ? response.data.message
+            : response.statusText
+    )
+    reporter.debugErr('Error details', response.data)
+}
+
+const resolveBundle = (cwd, params) => {
+    const appBundle = {}
+    // use file-path from params
+    if (params.file) {
+        try {
+            const filePath = path.resolve(cwd, params.file)
+            if (!fs.statSync(filePath).isFile()) {
+                reporter.error(`${params.file} is not a file`)
+                process.exit(1)
+            }
+            appBundle.id = params.appId
+            appBundle.version = params.fileVersion
+            appBundle.path = filePath
+            appBundle.name = path.basename(filePath)
+        } catch (e) {
+            reporter.error(`File does not exist at ${params.file}`)
+            process.exit(1)
+        }
+    } else {
+        // resolve file from built-bundle
+        const paths = makePaths(cwd)
+        const builtAppConfig = parseConfig(paths)
+        if (!builtAppConfig.id) {
+            reporter.error(
+                `No App Hub id found for app. Add an ${chalk.bold(
+                    'id'
+                )}-field to ${chalk.bold('d2.config.js')}`
+            )
+            process.exit(1)
+        }
+        appBundle.id = builtAppConfig.id
+        appBundle.version = builtAppConfig.version
+        appBundle.path = path.relative(
+            cwd,
+            paths.buildAppBundle
+                .replace(/{{name}}/, builtAppConfig.name)
+                .replace(/{{version}}/, builtAppConfig.version)
+        )
+        appBundle.name = builtAppConfig.name
+
+        if (!fs.existsSync(appBundle.path)) {
+            reporter.error(
+                `App bundle does not exist, run ${chalk.bold(
+                    'd2-app-scripts build'
+                )} before deploying.`
+            )
+            process.exit(1)
+        }
+    }
+
+    return appBundle
+}
+
+const promptForConfig = async params => {
+    if (process.env.CI && (!params.apikey || !params.minVersion)) {
+        reporter.error(
+            'Prompt disabled in CI mode - missing apikey or minVersion parameter.'
+        )
+        process.exit(1)
+    }
+
+    const responses = await inquirer.prompt([
+        {
+            type: 'input',
+            name: 'apikey',
+            message: 'App Hub API-key',
+            when: () => !params.apikey,
+        },
+        {
+            type: 'input',
+            name: 'minVersion',
+            message: 'Minimum DHIS2 version supported',
+            when: () => !params.minVersion,
+            validate: v =>
+                isValidServerVersion(v) ? true : 'Invalid server version',
+        },
+        {
+            type: 'input',
+            name: 'maxVersion',
+            message: 'Maximum DHIS2 version supported',
+            when: () => !params.maxVersion,
+            validate: v =>
+                !v || isValidServerVersion(v) ? true : 'Invalid server version',
```

Comment:
Maybe the DHIS `min` and `max` versions should be specified in `d2.config.js` instead of as command parameters :thinking: It would make publishing through CI easier as the publish command and its parameters run by the CI action would not have to change

Reply:
👍  agreed, I think that would be much cleaner.  We can keep this as-is for now and add that later if we want (@Birkbjo would you prefer to do it now or in a separate change?)

Reply:
I agree! That would be great, however I didn't want to make this dependent on adding support for that as well.

Reply:
Add an issue to the CLI project in Jira so we don't forget?